storage/range.go

```diff
@@ -507,19 +511,35 @@ func (r *Range) SetLastVerificationTimestamp(timestamp proto.Timestamp) error {
 // either along the read-only execution path or the read-write Raft
 // command queue.
 func (r *Range) AddCmd(ctx context.Context, call proto.Call) error {
-	args, reply := call.Args, call.Reply
+	args := call.Args
 	// TODO(tschottdorf) Some (internal) requests go here directly, so they
 	// won't be traced.
 	trace := tracer.FromCtx(ctx)
 	// Differentiate between admin, read-only and read-write.
+	var reply proto.Response
+	var err error
 	if proto.IsAdmin(args) {
 		defer trace.Epoch("admin path")()
-		return r.addAdminCmd(ctx, args, reply)
+		reply, err = r.addAdminCmd(ctx, args)
 	} else if proto.IsReadOnly(args) {
-		defer trace.Epoch("read path")()
-		return r.addReadOnlyCmd(ctx, args, reply)
+		defer trace.Epoch("read-only path")()
+		reply, err = r.addReadOnlyCmd(ctx, args)
+	} else if proto.IsWrite(args) {
+		defer trace.Epoch("read-write path")()
+		reply, err = r.addWriteCmd(ctx, args, nil)
+	} else {
+		panic("incomplete match")
```

Comment:
This is a strange error message. How about `panic(fmt.Sprintf("don't know how to handle command %T", args))`